extensions/mongodb/src/main/java/com/hazelcast/jet/mongodb/impl/ReadMongoParams.java

```diff
@@ -203,4 +208,52 @@ public ConnectorPermission buildPermissions() {
                 getCollectionName(),
                 ActionConstants.ACTION_READ);
     }
+
+    public static final class Aggregates implements Serializable {
+        private Document filter;
+        private Document projection;
+        private Document sort;
+
+        public List<Document> nonNulls() {
+            var list = new ArrayList<Document>();
+            if (filter != null) {
+                list.add(filter);
+            }
+            if (projection != null) {
+                list.add(projection);
+            }
+            if (sort != null) {
+                list.add(sort);
+            }
```

Comment:
Does Mongo not have some smart query planner and follow the order here? Is the fact that using filter first provides speedup documented somewhere?

Also, is this true in all scenarios? E.g what if there is a sorted index - wouldn't it be better to use sort first?

Reply:
I am not completely sure if it's true in every case. I saw many times that it's recommended to put the filters before everything, because some stages may make it impossible to push down the predicate. 6% improvement isn't much, but I think the code is even simpler to reason about, so I added the PR ;)